hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java

```diff
@@ -1872,6 +1872,21 @@ private int getWhenNoSSLDetected() {
     public static final HazelcastProperty PERSISTENCE_AUTO_CLUSTER_STATE_STRATEGY = new HazelcastProperty(
             "hazelcast.persistence.auto.cluster.state.strategy", ClusterState.NO_MIGRATION);
 
+    /**
+     * The absolute directory path to be used for jar uploading. The path must exist and must have
+     * read + write + execute permissions for the Hazelcast process
+     * <p>
+     * In case of a shared file system, the same absolute path can be specified for all cluster members,
+     * but this defeats the purpose of being able to distinguish jars uploaded to a specific member.
```

Comment:
This is actually even more confusing. Why should user want to upload anything on specific member directory in case the members use shared file system?

I would not complicate it and just provide the info whether:
- path can be different on each member
- path can be also relative